tests.py

```diff
@@ -7,6 +8,207 @@
+        input_shape = data_cf.input_shape
+        num_classes = data_cf.num_classes
+        preprocess_dataset_labels = data_cf.preprocess_dataset_labels
+        generate_new_model_for_dataset = data_cf.generate_new_model_for_dataset
+    if name == "mnist":
+        (x_train, y_train), (x_test, y_test) = mnist.load_data()
+        input_shape = data_mn.input_shape
+        num_classes = data_mn.num_classes
+        preprocess_dataset_labels = data_mn.preprocess_dataset_labels
+        generate_new_model_for_dataset = data_mn.generate_new_model_for_dataset
+
+    dataset = Dataset(
+            dataset_name=name,
+            x_train=x_train,
+            x_test=x_test,
+            y_train=y_train,
+            y_test=y_test,
+            input_shape=input_shape,
+            num_classes=num_classes,
+            preprocess_dataset_labels=preprocess_dataset_labels,
+            generate_new_model_for_dataset=generate_new_model_for_dataset
+            )
+    yield dataset
+
+
+class Test_Dataset:
+
+    def test_generate_new_model(self, create_Dataset):
+        assert create_Dataset.name in {"cifar10","mnist"}
+
+@pytest.fixture
+def create_partner_list(create_partner):
+    yield [create_partner] * 3
+
+
+@pytest.fixture
+def create_MultiPartnerLearning(create_Dataset, create_partner_list):
+    data = create_Dataset
+    part_list = create_partner_list
+    mpl = MultiPartnerLearning(
+            partners_list=part_list,
+            epoch_count=2,
+            minibatch_count=2,
+            dataset=data,
+            multi_partner_learning_approach="fedavg",
+            aggregation_weighting="uniform",
+            is_early_stopping=True,
+            is_save_data=False,
+            save_folder="",
+            )
+    yield mpl
+
+def test_mpl(create_MultiPartnerLearning):
+    assert type(create_MultiPartnerLearning) == MultiPartnerLearning
+
+
+
+@pytest.fixture(params=["cifar10", "mnist"])
+def create_scenario(request, create_MultiPartnerLearning, create_Dataset, create_partner_list):
+    dataset_name = request.param
+    params = {"dataset_name": dataset_name, "partners_count":3, "amounts_per_partner": [0.2, 0.5, 0.3], "samples_split_option": ["basic","random"], "multi_partner_learning_aproach":"fedavg", "aggregation_weighting": "uniform", "methods": ["Shapley values", "Independent scores"], "gradient_updates_per_pass_count": 5}
+
+    full_experiment_name = "unit-test-pytest"
+    experiment_path = Path.cwd() / "experiments" / full_experiment_name
+
+    scenar = Scenario(
+            params=params,
+            experiment_path=experiment_path,
+            scenario_id=0,
+            n_repeat=1
+            )
+
+    scenar.mpl = create_MultiPartnerLearning
+    scenar.dataset
+
+    yield scenar
+
+def test_scenar(create_scenario):
+    assert type(create_scenario) == Scenario
+
+@pytest.fixture(scope="class")
+def create_cifar10_x_train():
+    (x_train, y_train), (x_test, y_test) = cifar10.load_data()
```

Comment:
Do you need `(x_test, y_test)` here, as it seems to be unused?

Reply:
or maybe test the `train` and `test`?

Reply:
To create `x_train` I have to load these 3 variables, I could also make test on `x_test`

I started writing the class `Test_dataset_cifar10` to learn `pytest` - I could maybe keep this class as an example
I will try to implement test on the labels `y_train` and `y_test`.

Reply:
Ok great!